### DISCUSSION TITLE
Env var

### DISCUSSION THREAD

**Discussion by plum59** [argoproj/argo-rollouts#2339] [Q&A]
Hello  all,
I have specific querry and i don't know how manage with this.

To explain:  i have an application that talks with another application and so that the second can return its call back, the first sends it with an environment variable the name of the service and the port on which it must make its call back. 
For the blue/green deployment to work, the second application must therefore call back on the preview or active service. except that the argo rollout method does not allow on-the-fly modification of environment variables.

there is definitely a better way to do it, any ideas are welcome.

**Comment by kostis-codefresh** [comment]
Change the second application to simply have a static settings called "callback:http://firstservice:999" or something similar.
Then create a K8s service called "firstservice" and make it managed by Argo Rollouts

Then during deployments "firstservice" will automatically be switched between active/preview by argo rollouts and no modifications need to happen anywhere else.

**Reply by plum59** [reply]
Hello in fist , thx you

Unless I follow you correctly we would have same issu.

During the tests when the two replicaset will be up:
During a call on the active app  the second application will answer on preview service , so on preview app and during a call on the preview app also.
which will not allow us to test the new preview applications from end to end (with call back) and in same time  keeping the active application completely available for our customer.

Before rollout
![image](https://user-images.githubusercontent.com/57859673/196889530-cf14797f-f5b6-4616-91d4-afb4f7b7ee98.png)

During rollout process
![image](https://user-images.githubusercontent.com/57859673/196889596-53b092b6-13f5-473c-9db2-d66abc0c13c5.png)

So my point is realy about during rollout process to have avability to keep application completely available for customer and in same time have avability to test application end to end.

**Reply by kostis-codefresh** [reply]
So I haven't worked with all traffic providers in Argo Rollouts but for several cases (ambassador, nginx) Argo Rollouts creates 3 services during rollout

1) stable => this service sends traffic ALWAYS to previous version
2) preview => this service sends traffic ALWAYS to new version
3) canary => This service sends traffic either to new or old version according to weight

So it depends on what your business requirements are. If you want App2 to always be stable then set your setting to "http://stable:9999". If you want to always test with new version (even when canary is at 10%) then setting to "http://preview:9999". And if you want App 2 to follow what App1 does then use setting "http://canary:9999"

I hope that helps.

**Reply by plum59** [reply]
Thx a lot i will check it ty
